packages/cli/src/cli/config/config.diff.ts

```diff
@@ -51,3 +60,100 @@ export class ConfigDiff {
     return false;
   }
 }
+
+/**
+ * Given a old and new lds layer stac item and log the changes
+ */
+export function getVectorChanges(newLayer: StacLink | undefined, existingLayer: StacLink | undefined): string | null {
+  // Update Layer
+  if (newLayer != null && existingLayer != null) {
+    const featureChange = Number(newLayer['lds:feature_count']) - Number(existingLayer['lds:feature_count']);
+
+    if (newLayer['lds:version'] === existingLayer['lds:version'] && featureChange !== 0) {
+      // Alert if feature changed with no version bump.
+      return `🟥🟥🟥🟥 Feature Change Detected ${newLayer['lds:name']} - version: ${newLayer['lds:version']} features: ${newLayer['lds:feature_count']} (+${featureChange}) 🟥🟥🟥🟥`;
+    }
+
+    if (featureChange >= 0) {
+      // Add Features
+      return `🟦 ${newLayer['lds:name']} - version: ${newLayer['lds:version']} (from: ${existingLayer['lds:version']}) features: ${newLayer['lds:feature_count']} (+${featureChange})`;
+    } else {
+      // Remove Features
+      return `🟧 ${newLayer['lds:name']} - version: ${newLayer['lds:version']} (from: ${existingLayer['lds:version']}) features: ${newLayer['lds:feature_count']} (-${featureChange})`;
+    }
+  }
+
+  // Add new Layer
+  if (newLayer != null && existingLayer == null) {
+    return `🟩 ${newLayer['lds:name']} - version: ${newLayer['lds:version']} features: ${newLayer['lds:feature_count']}`;
+  }
+
+  // Remove Layer
+  if (newLayer == null && existingLayer != null) {
+    return `🟥 ${existingLayer['lds:name']} features: -${existingLayer['lds:feature_count']}`;
+  }
+
+  // No changes detected return null
+  return null;
+}
+
+/**
+ * Prepare and create pull request for the aerial tileset config
+ */
+export async function diffVectorUpdate(
+  tileSet: ConfigTileSet,
+  existingTileSet: ConfigTileSet | null,
+): Promise<string | undefined> {
+  // Vector layer only support for 3857 and only contain on layer inside
+  const changes: string[] = [];
+  const layer = tileSet.layers[0];
+  if (layer[Epsg.Google.code] == null) return;
```

Comment:
could layer be null?

```suggestion
  if (layer?.[Epsg.Google.code] == null) return;
```

Reply:
I was thinking we might want to error here? Because we don't support other projections in vector map and we shouldn't have empty layers in a tileset

Reply:
If we want the error we should throw, having "Cannot read property of undefined" is not super helpful

Reply:
Just update this to thrown error.